[Maintenance] Removes `mimimatch` manual resolution

tmarkley opened the pull request:
### Description
* The `minimatch` resolution was no longer necessary because an upstream library that depended on `minimatch@3.0.4` was removed in #2711.
 
### Issues Resolved
N/A - related to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1298
 
### Check List
- [x] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

AMoo-Miki commented on yarn.lock:
```diff
@@ -12686,13 +12659,27 @@ minimalistic-crypto-utils@^1.0.1:
+  dependencies:
+    brace-expansion "^2.0.1"
+
+minimatch@~3.0.4:
```
Do we want to push this to `3.1.2` like before?

tmarkley replied:
I don't think we need to add a manual resolution in this case since `3.0.8` addresses the CVE that necessitated the bump to `3.0.5` previously. That could have been addressed without a manual resolution by bumping the patch version in the yarn.lock file (remove the version and then re-bootstrap).

ashwin-pc replied:
@tmarkley @miki, not related to this specific PR, but are there reference docs to understand what these changes are and how we need to apply them w.r.t OSD?  Since the yarn lock file is an auto generated file and usually automatically managed, I am not too familiar as to why the inital manual resolution was necessary and why we are removing it now. I understand your reasoning for the change @tmarkley but i do not know enough to know:
1. Why we need to manage this manually
2. Why we do it the way we do it today.

Having a common understanding of its workings will help others who have to upgrade the dependencies correctly too.

tmarkley replied:
@ashwin-pc yes, I'm going open a PR to include that kind of information in the SECURITY file for managing CVEs and dependencies. If you look at the commit history/git blame, you can see that the resolution was added in #2640.

Overall, it is correct that the yarn.lock file should not be modified manually, but adding [selective dependency resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions) can cause issues with certain libraries due to compatibility issues (e.g. if a library in the package.json depends on an older version of an upstream library that was resolved to a newer version that breaks previous functionality). I think it's best to strive to avoid resolutions whenever possible and only reserve them for situations where we're resolving a CVE and have a longer-term plan to upgrade/remove/change dependencies and then remove the resolution once that work is done.

AMoo-Miki reviewed (changes requested):
Can you please check the test failures [here](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/3606795687/jobs/6078389527)? To me, it appears like a bad snapshot. Would you like to use `expect.any(Number)` instead of those hardcoded numbers?

tmarkley commented:
```
$ yarn why minimatch
yarn why v1.22.15
[1/4] Why do we have the module "minimatch"...?
[2/4] Initialising dependency graph...
warning Resolution field "typescript@4.0.2" is incompatible with requested version "typescript@~4.5.2"
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "minimatch@3.1.2"
info Has been hoisted to "minimatch"
info Reasons this module exists
   - "workspace-aggregator-450caac8-e5d2-4eac-8cc3-0592515f6a48" depends on it
   - Specified in "dependencies"
   - Hoisted from "_project_#glob#minimatch"
   - Hoisted from "_project_#eslint#minimatch"
   - Hoisted from "_project_#eslint-plugin-import#minimatch"
   - Hoisted from "_project_#eslint-plugin-jsx-a11y#minimatch"
   - Hoisted from "_project_#eslint-plugin-node#minimatch"
   - Hoisted from "_project_#minimatch"
   - Hoisted from "_project_#eslint-plugin-react#minimatch"
   - Hoisted from "_project_#tape#dotignore#minimatch"
   - Hoisted from "_project_#load-grunt-config#glob#minimatch"
   - Hoisted from "_project_#@osd#pm#multimatch#minimatch"
   - Hoisted from "_project_#archiver#readdir-glob#minimatch"
   - Hoisted from "_project_#nyc#test-exclude#minimatch"
   - Hoisted from "_project_#grunt#findup-sync#glob#minimatch"
   - Hoisted from "_project_#@osd#plugin-generator#ejs#jake#minimatch"
   - Hoisted from "_project_#@osd#plugin-generator#ejs#jake#filelist#minimatch"
info Disk size without dependencies: "44KB"
info Disk size with unique dependencies: "64KB"
info Disk size with transitive dependencies: "124KB"
info Number of shared dependencies: 3
=> Found "mocha#minimatch@5.0.1"
info This module exists because "_project_#mocha" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "72KB"
info Disk size with transitive dependencies: "132KB"
info Number of shared dependencies: 3
=> Found "grunt#minimatch@3.0.8"
info This module exists because "_project_#grunt" depends on it.
info Disk size without dependencies: "44KB"
info Disk size with unique dependencies: "64KB"
info Disk size with transitive dependencies: "124KB"
info Number of shared dependencies: 3
=> Found "grunt#glob#minimatch@3.1.2"
info Reasons this module exists
   - "_project_#grunt#glob" depends on it
   - Specified in "dependencies"
info Disk size without dependencies: "44KB"
info Disk size with unique dependencies: "64KB"
info Disk size with transitive dependencies: "124KB"
info Number of shared dependencies: 3
=> Found "@osd/ui-framework#minimatch@3.1.2"
info Reasons this module exists
   - "_project_#@osd#ui-framework#grunt-contrib-clean#rimraf#glob" depends on it
   - Specified in "dependencies"
   - Hoisted from "_project_#@osd#ui-framework#grunt-contrib-clean#rimraf#glob#minimatch"
   - in the nohoist list ["/_project_/**/@types/*","/_project_/**/@types/*/**","/_project_/**/grunt-*","/_project_/**/grunt-*/**","/_project_/@elastic/eui/rehype-react","/_project_/@elastic/eui/remark-rehype","/_project_/@elastic/eui/remark-rehype/**"]
info Disk size without dependencies: "44KB"
info Disk size with unique dependencies: "64KB"
info Disk size with transitive dependencies: "124KB"
info Number of shared dependencies: 3
Done in 1.42s.

$ npm list minimatch
opensearch-dashboards@3.0.0 /workspaces/OpenSearch-Dashboards
├─┬ @osd/plugin-generator@1.0.0 -> /workspaces/OpenSearch-Dashboards/packages/osd-plugin-generator
│ └─┬ ejs@3.1.7
│   └─┬ jake@10.8.5
│     ├─┬ filelist@1.0.2
│     │ └── minimatch@3.1.2  deduped
│     └── minimatch@3.1.2  deduped
├─┬ archiver@5.3.0
│ └─┬ readdir-glob@1.1.1
│   └── minimatch@3.1.2  deduped
├─┬ eslint@6.8.0
│ └── minimatch@3.1.2  deduped
├─┬ eslint-plugin-import@2.25.4
│ └── minimatch@3.1.2  deduped
├─┬ eslint-plugin-jsx-a11y@6.5.1
│ └── minimatch@3.1.2  deduped
├─┬ eslint-plugin-node@11.1.0
│ └── minimatch@3.1.2  deduped
├─┬ eslint-plugin-react@7.29.4
│ └── minimatch@3.1.2  deduped
├─┬ glob@7.2.0
│ └── minimatch@3.1.2  deduped
├─┬ grunt@1.5.3
│ ├─┬ findup-sync@0.3.0
│ │ └─┬ glob@5.0.15
│ │   └── minimatch@3.1.2  deduped
│ ├─┬ glob@7.1.7
│ │ └── minimatch@3.1.2 
│ └── minimatch@3.0.8 
├─┬ load-grunt-config@4.0.1
│ ├─┬ glob@7.1.7
│ │ └── minimatch@3.1.2  deduped
│ └─┬ load-grunt-tasks@5.1.0
│   └─┬ multimatch@4.0.0
│     └── minimatch@3.1.2  deduped
├── minimatch@3.1.2 
├─┬ mocha@10.1.0
│ └── minimatch@5.0.1 
├─┬ nyc@15.1.0
│ └─┬ test-exclude@6.0.0
│   └── minimatch@3.1.2  deduped
└─┬ tape@5.5.2
  └─┬ dotignore@0.1.2
    └── minimatch@3.1.2  deduped
```

AMoo-Miki reviewed (changes requested):
Looks like `globby@11.1.0` is not compatible with `@osd/dev-utils`.

tmarkley commented:
> Looks like `globby@11.1.0` is not compatible, at least, with `@osd/dev-utils`.

Yeah, looks like that will require some additional investigation. I've removed those changes and simplified.

joshuarrrr commented:
@tmarkley This is not backportable, correct? At least a quick glance at #2711 says it's labeled for `3.0.0`.

tmarkley commented:
> @tmarkley This is not backportable, correct? At least a quick glance at #2711 says it's labeled for `3.0.0`.

@joshuarrrr Correct.
